person.rb

```diff
@@ -0,0 +1,21 @@
+class Person
+  attr_reader :@id
+  attr_accessor :name, :age
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..100)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def of_age?
+    return true if @age >= 18
+  end
+
+  def can_use_services?
+    return true if of_age? or @parent_permission == true
+  end
+
+  private :of_age?
```

Comment:
**[OPTIONAL]**

- [ ] It would be nice if you make these both methods using less code to make your app more efficient and work faster such as below ⬇️

```ruby

  def of_age?
    @age >= 18
  end

  def can_use_services? 
    of_age?() || @parent_permission
  end

 private :of_age?  
```

Reply:
@HAMMAS-SALEEM Done. Thanks